lib/engine.py

```diff
@@ -264,32 +260,38 @@ def auto_get_delimiter(self, header):
 
     def convert_data_type(self, datatype):
         """Converts Retriever generic data types to database platform specific
-        data types"""
-        thistype = datatype[0]
+        data types
+        """
+        # get the type from the dataset variables
+        key = datatype[0]
         thispk = False
-        if thistype[0:3] == "pk-":
-            thistype = thistype.lstrip("pk-")
+        if key[0:3] == "pk-":
+            key = key.lstrip("pk-")
             thispk = True
-        elif thistype[0:3] == "ct-":
-            thistype = thistype[3:]
-
-        if thistype in list(self.datatypes.keys()):
-            thistype = self.datatypes[thistype]
+        elif key[0:3] == "ct-":
+            key = key[3:]
```

Comment:
Oops, never mind. I was reading past `thispk = True`. Still probably best to change the `lstrip` to `key[3:]`.